src/messagingService/index.js

```diff
@@ -9,39 +11,45 @@ export const checkForPermissions = async () => {
     try {
       await messaging().requestPermission()
     } catch (error) {
-      console.log(error.message)
+      throw new MessagingError(error.message)
     }
   }
 }
 
 export const saveFCMTokenForCurrentUser = async () => {
   if (messaging().hasPermission()) {
     const fcmToken = await messaging().getToken()
-
     if (fcmToken) {
       try {
+        await CacheService.setItem('fcmToken', fcmToken)
         const storeFCMTokenInUserGroup = firebase.functions().httpsCallable('storeFCMTokenInUserGroup')
 
         await storeFCMTokenInUserGroup({ fcmToken: fcmToken })
       } catch (error) {
-        console.log(error.message)
+        throw new MessagingError(error.message)
       }
     }
   }
 }
 
 export const deleteFCMTokenForCurrentUser = async () => {
-  if (await messaging().hasPermission()) {
-    const fcmToken = await messaging().getToken()
+  let fcmToken
 
-    if (fcmToken) {
-      try {
-        const removeFCMTokenInUserGroup = firebase.functions().httpsCallable('removeFCMTokenInUserGroup')
+  try {
+    fcmToken = await CacheService.getItem('fcmToken')
 
-        await removeFCMTokenInUserGroup({ fcmToken: fcmToken })
-      } catch (error) {
-        console.log(error.message)
-      }
+    await CacheService.removeItem('fcmToken')
+  } catch (error) {
+    fcmToken = null
+  }
+
+  if (fcmToken) {
```

Comment:
If I remember correctly, there's a rule in Clean Code that forbids doing this. You're basically checking if it didn't fail to continue with the rest of the code. You could've fetched the token, and do the rest only if a token exists:

```javascript
const fcmToken = await CacheService.getItem('fcmToken')

if (!fcmToken) return undefined

try {
  // remove the tokem from the cache and FCM group
} catch (err) {
  // throw an appropriate error
}
```

Don't change it now tho, we need to advance with new features